src/main/java/com/uber/cadence/internal/replay/ReplayDecider.java

```diff
@@ -418,8 +420,17 @@ private void decideImpl(PollForDecisionTaskResponse decisionTask, Functions.Proc
         decisionsHelper.handleDecisionTaskStartedEvent(decision);
       }
     } catch (Error e) {
-      metricsScope.counter(MetricsType.DECISION_TASK_ERROR_COUNTER).inc(1);
-      throw e;
+      if (this.workflow.getWorkflowImplementationOptions().getNonDeterministicWorkflowPolicy()
```

Comment:
Should we check if the error is NonDeterminisicWorkflowError?

Reply:
I think any Error should be treated this way. I kept the constant name similar to Go version for consistency.